File: calmar-ocean-op-java/src/main/java/ie/marei/calmar/OwiParameters.java
package ie.marei.calmar;

import org.esa.snap.core.datamodel.Band;
import org.esa.snap.core.datamodel.Product;
import org.esa.snap.core.gpf.OperatorException;

public class OwiParameters {

    private final String[][] possibleOwiParameterNames = {
            {"vv_001_owiWindSpeed", "vv_001_owiLat", "vv_001_owiLon"},
            {"hh_001_owiWindSpeed", "hh_001_owiLat", "hh_001_owiLon"}
    };

    private Product sourceProduct;
    private Band windBand;
    private double noData = 0.0;
    private String owiWindSpeedName = "";
    private String owiLatName = "";
    private String owiLonName = "";

    public OwiParameters(Product sourceProduct) {

        if (sourceProduct == null) {
            throw new OperatorException("Source product is missing");
        }
        this.sourceProduct = sourceProduct;

        Band wind = null;
        for (String[] parameters : this.possibleOwiParameterNames) {
            Band checkBand = sourceProduct.getBand(parameters[0]);
            if (checkBand != null) {
                this.owiWindSpeedName = parameters[0];
                this.owiLatName = parameters[1];
                this.owiLonName = parameters[2];
                wind = checkBand;
                break;
            }
        }
        if (wind == null) {
            throw new OperatorException("Requires a Sentinel 1 Level 2 OCN source product");
        }
        this.noData = wind.getGeophysicalNoDataValue();
        this.windBand = wind;
    }

    public Band getWindBand() {
        return windBand;
    }

    public double getNoData() {
        return noData;
    }

    public String getOwiWindSpeedName() {
        return owiWindSpeedName;
    }

    public String getOwiLatName() {
        return owiLatName;
    }

    public String getOwiLonName() {
        return owiLonName;
    }
}

File: calmar-ocean-op-python/src/main/python/owi_parameters.py
import numpy
from snappy import jpy, Band, ProductIO, ProductUtils

class OwiParameters:
    _possibleOwiParameterNames = [["vv_001_owiWindSpeed", "vv_001_owiLat", "vv_001_owiLon"],
                                  ["hh_001_owiWindSpeed", "hh_001_owiLat", "hh_001_owiLon"]]

    def __init__(self, source_product):
        self._source_product = None
        self._wind_band = None
        self._no_data = 0.0
        self._owi_wind_speed_name = ""
        self._owi_lat_name = ""
        self._owi_lon_name = ""

        if source_product is None:
            raise RuntimeError("Source product is missing")
        self._source_product = source_product

        wind = None
        for parameters in self._possibleOwiParameterNames:
            check_band = source_product.getBand(parameters[0])
            if check_band is not None:
                self._owi_wind_speed_name = parameters[0]
                self._owi_lat_name = parameters[1]
                self._owi_lon_name = parameters[2]
                wind = check_band
                break
        if wind is None:
            raise RuntimeError("Requires a Sentinel 1 Level 2 OCN source product")
        self._no_data = wind.getGeophysicalNoDataValue()
        self._wind_band = wind

    def get_wind_band(self):
        return self._wind_band

    def get_no_data(self):
        return self._no_data

    def get_owi_wind_speed_name(self):
        return self._owi_wind_speed_name

    def get_owi_lat_name(self):
        return self._owi_lat_name

    def get_owi_lon_name(self):
        return self._owi_lon_name
